frontend/src/main/scala/bloop/bsp/BspServer.scala

```diff
@@ -245,9 +246,14 @@ object BspServer {
     val deleteExternalDirsTask = latestState.build.projects.map { project =>
       import bloop.io.Paths
       import java.io.IOException
-      val externalClientClassesDir = latestState.client.getUniqueClassesDirFor(project)
-      if (externalClientClassesDir == project.genericClassesDir) Task.now(())
-      else Task.eval(Paths.delete(externalClientClassesDir)).executeWithFork
+      try {
+        val externalClientClassesDir = latestState.client.getUniqueClassesDirFor(project)
+        if (externalClientClassesDir == project.genericClassesDir) Task.now(())
+        else Task.fork(Task.eval(Paths.delete(externalClientClassesDir)))
+      } catch {
+        // Can be thrown from `getUniqueClassesDirFor`
+        case _: NoSuchFileException => Task.now(())
```

Comment:
Can't we check if that file exists instead of the try?

Reply:
It's not as trivial as doing a check for existence, that was the first approach I tried and it didn't work, but you're right this falls short. I'll rebase this PR on top of the PR changing lots of things re classes directories and try to come up with a better solution.